### ISSUE TITLE
Can't get the spectrum integrated with OpenMP

### ISSUE THREAD

**Issue by epassaro** [tardis-sn/tardis#1019]
**SO:** Ubuntu 18.04
**Linux Kernel:** 5.3.0-26-generic
**Compilers:** using the latest `conda-forge` compilers
**TARDIS version:** 3.0.dev3378

**Steps:** 
1. Install via TARDIS via `setup.py` with the `--with-openmp` flag.
2. Run simulation with default `tardis_example.yml` and default atomic data, with `nthreads` greater than 1 and call `simulation.runner.spectrum_integrated` lazy attribute.

**Output:**
```
[tardis.simulation.base][INFO   ]  Simulation finished in 20 iterations and took 179.66 s (base.py:308)
Doing the formal integral
Running with OpenMP - 4 threads
        [100%] Bins(finished/total): 10000/10000
double free or corruption (!prev)
Makefile:4: recipe for target 'spectra' failed
make: *** [spectra] Aborted (core dumped)
```

**Comment by epassaro** [comment]
Maybe this is related to the changes introduced in #995

**Comment by epassaro** [comment]
I can confirm this bug was introduced in #995 @wkerzendorf

**Comment by Chinmay4400** [comment]
@epassaro I tried recreating the issue with these steps : 
installed tardis with --with-openmp
changed tardis_example.yml file with nthreads=3 and method : integrated in spectrum block

However upon running the quickstart notebook,  the simulation simply ended with 
```[tardis.simulation.base][INFO   ]  Simulation finished in 20 iterations and took 36.05 s (base.py:308)
```
Am I executing all necessary steps?

**Comment by Phantsure** [comment]
I tried reversing the changes in #995, still same error. My machine specs are similar.

**Comment by kaushik94** [comment]
I think #995 should be reverted anyways @epassaro @wkerzendorf .

**Comment by epassaro** [comment]
Fixed by #1159.
